notebooks/302-pytorch-quantization-aware-training/302-pytorch-quantization-aware-training.ipynb

```diff
@@ -28,9 +28,42 @@
     "id": "6M1xndNu-z_2"
    },
    "source": [
-    "# Settings\n",
+    "# Imports and Settings\n",
+    "\n",
+    "On Windows, add the required C++ directories to the system's path.\n",
+    "\n",
     "Import NNCF and all auxiliary packages from your Python* code.\n",
-    "Set a name for the model, and the image width and height that will be used for the network. Also define paths where PyTorch, ONNX and OpenVINO IR versions of the models will be stored."
+    "Set a name for the model, and the image width and height that will be used for the network. Also define paths where PyTorch, ONNX and OpenVINO IR versions of the models will be stored. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# On Windows, add the directory that contains cl.exe to the PATH to enable PyTorch to find the \n",
+    "# required C++ tools. This code assumes that Visual Studio 2019 is installed in the default \n",
+    "# directory. If you have a different C++ compiler, please add the correct path to os.environ[\"PATH\"] \n",
+    "# directly.\n",
+    "\n",
+    "import sys\n",
+    "import subprocess\n",
+    "if sys.platform == \"win32\" and subprocess.run([\"where\", \"cl\"]).returncode == 1:\n",
+    "    from pathlib import Path\n",
+    "    import os\n",
+    "    VS_INSTALL_DIR = r\"C:/Program Files (x86)/Microsoft Visual Studio\"\n",
```

Comment:
Have you considered [vswhere](https://docs.microsoft.com/en-us/visualstudio/install/tools-for-managing-visual-studio-instances?view=vs-2019) tool instead? Potentially, it can be more general.


> vswhere.exe` is automatically included in Visual Studio 2017

Reply:
I did! I came across it thinking surely there must be a better way. But I found it not very intuitive, it is only included from VS 2017+ (I have no idea how common VS 2015 still is - that may not be an issue) and you would still need to assume the installation in C:\Program files (x86). Or you would have to download the vswhere executable in the notebook, but that seemed excessive and potentially triggering security flags. If you have a simple suggestion for how to use `vswhere` here I would be happy to use that. 

I also think there is a smarter way even with path searching, but I did not want to make the code in this cell even longer just to do this check for Windows users. Improvements welcome though!